### DISCUSSION TITLE
How to use pure Rust language and Rust-GPU to modify the Bevy_render crate instead of using wgpu?

### DISCUSSION THREAD

**Discussion by BlackHatChen** [bevyengine/bevy#7369] [Q&A]
Hello, I'm recently trying to work on using Rust-GPU and pure Rust language with Bevy_render crate, but not sure how to modify it.
Like camera_driver_node.rs:
![image](https://user-images.githubusercontent.com/86137974/214795715-2706460f-8fcf-4018-bf55-80156fd63436.png)

I know the part of code highlighted with blue marks is making the window's screen being all black if there's no camera include in main function, and that just makes me reminding the syntax: ClearColor(Color::rgb(BLACK)) which used in App::new() in main function, but I don't know how to use in crate, or not sure whether it can be used in this place.

In brief, I want to replace wgpu syntax, but not found the suitable syntax for wgpu in rust.
Or maybe I have a mistake? Like instead of  "just replace the syntax", I have to "rewrite all the bevy_render crate (except for macros)"?

**Comment by bravely-beep** [comment]
I'm not really sure what you're trying to achieve here. `wgpu` is a graphics API, and `rust-gpu` is a shader compiler; they each do different, separate things. (The examples in the `rust-gpu` repo show how the two crates can be used together as two separate parts of a larger program).

**Reply by soqb** [reply]
@YuOAO-dreamlife are you confusing [`wgpu`](https://github.com/gfx-rs/wgpu) (the pure Rust rendering library crate) with [wgsl](https://www.w3.org/TR/WGSL/) (the shader language)? confusingly both are in and around [WebGPU](https://www.w3.org/TR/webgpu/) (an experimental graphics api to be used on the web) which is the replacement for [WebGL](https://www.khronos.org/webgl/).

**Reply by BlackHatChen** [reply]
@bravely-beep
@soqb 
Sorry it's taken me so long to get back to you, I'm sure that I was mistaken with `wgpu`. 
But now I have new question, is it possible to write `Bevy-render` crate **without using API**?
Just use Rust language and compile with `rust-gpu`, turn it directly into SPIR-V?
